<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.29">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Fabrício Almeida-Silva">
<meta name="dcterms.date" content="2023-06-12">
<meta name="description" content="Check out where Bioc developers have published their papers">

<title>Where can I publish a paper describing my Bioconductor package? – Fabrício Almeida-Silva</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../img/logo.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-0815c480559380816a4d1ea211a47e91.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-c5cb712cbad652d70ce516660dbb968f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-9TMPT34HZ4"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-9TMPT34HZ4', { 'anonymize_ip': true});
</script>
<script src="../../site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.27/datatables.js"></script>
<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="../../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<link rel="stylesheet" href="../../styles.css">
<meta name="twitter:title" content="Fabrício Almeida-Silva">
<meta name="twitter:description" content="Hi, I’m Fabrício! Welcome to my personal website and blog.">
<meta name="twitter:image" content="https://almeidasilvaf.github.io/blog/2022-01-03-bioc_publications/bioc.gif">
<meta name="twitter:creator" content="@almeidasilvaf">
<meta name="twitter:site" content="@almeidasilvaf">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../img/logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Fabrício Almeida-Silva</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about/index.html"> 
<span class="menu-text">About me</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../software/index.html"> 
<span class="menu-text">Software &amp; Data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../publications/index.html"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../teaching/index.html"> 
<span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../blog/index.html"> 
<span class="menu-text">Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://bsky.app/profile/almeidasilvaf.bsky.social"> 
<span class="menu-text"><i class="fa-brands fa-bluesky" aria-label="bluesky"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/almeidasilvaf"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:fabricio_almeidasilva@hotmail.com"> <i class="bi bi-envelope" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#motivation" id="toc-motivation" class="nav-link active" data-scroll-target="#motivation">Motivation</a></li>
  <li><a href="#extracting-citation-information-from-bioconductors-browsable-code-base" id="toc-extracting-citation-information-from-bioconductors-browsable-code-base" class="nav-link" data-scroll-target="#extracting-citation-information-from-bioconductors-browsable-code-base">Extracting citation information from Bioconductor’s browsable code base</a></li>
  <li><a href="#summary-stats" id="toc-summary-stats" class="nav-link" data-scroll-target="#summary-stats">Summary stats</a></li>
  <li><a href="#session-information" id="toc-session-information" class="nav-link" data-scroll-target="#session-information">Session information</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Where can I publish a paper describing my Bioconductor package?</h1>
  <div class="quarto-categories">
    <div class="quarto-category">bioinformatics</div>
    <div class="quarto-category">reproducible research</div>
    <div class="quarto-category">bioconductor</div>
    <div class="quarto-category">scientific writing</div>
    <div class="quarto-category">rstats</div>
  </div>
  </div>

<div>
  <div class="description">
    Check out where Bioc developers have published their papers
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Fabrício Almeida-Silva <a href="https://orcid.org/0000-0002-5314-2964" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 12, 2023</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="motivation" class="level2">
<h2 class="anchored" data-anchor-id="motivation">Motivation</h2>
<p>When I developed <a href="https://bioconductor.org/packages/release/bioc/html/BioNERO.html">BioNERO</a>, my first R/Bioconductor package, I didn’t know to which journals I could submit the paper describing it. Since then, I’ve seen many other R developers that have faced the same issue. To help solve this problem, here I will guide you on how to do some web scraping to find out the main journals where Bioc developers publish papers describing their packages.</p>
</section>
<section id="extracting-citation-information-from-bioconductors-browsable-code-base" class="level2">
<h2 class="anchored" data-anchor-id="extracting-citation-information-from-bioconductors-browsable-code-base">Extracting citation information from Bioconductor’s browsable code base</h2>
<p>Bioconductor offers a <a href="https://code.bioconductor.org/">browsable code base</a> that lets users explore git repositories and search code in all Bioconductor packages. If we go to the <strong>Code Search</strong> page and search <code>journal f:CITATION</code>, we will get a list of all CITATION files (where developers include citation information for their packages) that include the string “journal”.</p>
<p>Knowing that, we can do some web scraping using the <em><a href="https://CRAN.R-project.org/package=rvest">rvest</a></em> package to extract such information for all packages and parse it into a nicely-formatted data frame.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load required packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rvest)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get URL of the search "journal f:CITATION"</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://code.bioconductor.org/search/search?q=journal%20f%3aCITATION"</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">2000</span> <span class="co"># number of files to show</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Get list of tables containing journal names</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>journal_list <span class="ot">&lt;-</span> rvest<span class="sc">::</span><span class="fu">read_html</span>(<span class="fu">paste0</span>(url, <span class="st">"&amp;num="</span>, n)) <span class="sc">|&gt;</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>    rvest<span class="sc">::</span><span class="fu">html_table</span>()</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Parse list of data frames into a large, tidy list</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>journal_df <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">lapply</span>(<span class="fu">seq_along</span>(journal_list), <span class="cf">function</span>(x) {</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> journal_list[[x]]</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Package name</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>    pkg <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">":.*"</span>, <span class="st">""</span>, <span class="fu">names</span>(df)[<span class="dv">1</span>])</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(df) <span class="ot">&lt;-</span> <span class="st">"entries"</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df) <span class="sc">|&gt;</span> </span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 1) Keep only rows containing 'journal=' or 'journal ='</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(entries, <span class="st">"journal</span><span class="sc">\\</span><span class="st">s*="</span>)) <span class="sc">|&gt;</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 2) Get journal name (remove quotation marks, whitespace, commas, etc)</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(entries, <span class="st">".*="</span>, <span class="st">""</span>),</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">'</span><span class="sc">\\\"</span><span class="st">'</span>, <span class="st">''</span>),</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">"'"</span>, <span class="st">''</span>),</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">""</span>),</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_squish</span>(journal),</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_to_upper</span>(journal),</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">",$"</span>, <span class="st">""</span>),</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">"</span><span class="sc">\\</span><span class="st">)"</span>, <span class="st">""</span>),</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">"</span><span class="sc">\\</span><span class="st">("</span>, <span class="st">""</span>),</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">"</span><span class="sc">\\</span><span class="st">{"</span>, <span class="st">""</span>),</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="st">"</span><span class="sc">\\</span><span class="st">}"</span>, <span class="st">""</span>)</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">|&gt;</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(journal)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a column named `package` containing package name</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(df) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">package =</span> pkg)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Taking a look at the first rows</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(journal_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             journal           package
1     BIOINFORMATICS        cytomapper
2 SCIENTIFIC REPORTS      IsoCorrectoR
3     BIOINFORMATICS             Rtpca
4     BIOINFORMATICS transcriptogramer
5     BIOINFORMATICS               ACE
6     BIOINFORMATICS          limmaGUI</code></pre>
</div>
</div>
<p>Now, because CITATION files are created manually by developers, a big (and expected) problem is the lack of standardization. This leads to different developers referring to the same journal by different names (e.g., Nature Methods and Nat Methods, Nucleic Acids Research and NAR, etc). You can see that yourself by executing <code>sort(unique(journal_df$journal))</code>. While I can never expect to fix this problem completely (especially if you are reading this post in the future and new packages have been added), below is my attempt to fix most of the inconsistencies. I will probably miss some strange exceptions, but I guess I can live with it, right?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 'Journals' to remove (these are not actually journals)</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>to_remove <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">""</span>, <span class="st">"07"</span>, <span class="st">"1"</span>, <span class="st">"10"</span>, <span class="fu">as.character</span>(<span class="dv">2010</span><span class="sc">:</span><span class="dv">2023</span>), <span class="st">"2022-2032"</span>,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IN REVIEW"</span>, <span class="st">"IN PREPARATION"</span>, <span class="st">"JOURNAL"</span>, <span class="st">"MANUSCRIPT IN PREPARATION"</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TBA"</span>, <span class="st">"TBD"</span>, <span class="st">"UNDER REVIEW"</span>, <span class="st">"UNIVERSITY OF REGENSBURG"</span>,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"BIOCONDUCTOR"</span>, <span class="st">"SUBMITTED"</span>, <span class="st">"MEDRXIV"</span>, <span class="st">"BIORXIV"</span>, <span class="st">"PREPRINT"</span>, <span class="st">"ARXIV"</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Standardize names</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>journal_df_clean <span class="ot">&lt;-</span> journal_df <span class="sc">|&gt;</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>journal <span class="sc">%in%</span> to_remove) <span class="sc">|&gt;</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">journal =</span> <span class="fu">str_replace_all</span>(journal, <span class="fu">c</span>(</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ALBANY NY.*"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ALGORITHMS MOL BIO"</span> <span class="ot">=</span> <span class="st">"ALGORITHMS FOR MOLECULAR BIOLOGY"</span>,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ANAL CHEM"</span> <span class="ot">=</span> <span class="st">"ANALYTICAL CHEMISTRY"</span>,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ANN APPL STAT"</span> <span class="ot">=</span> <span class="st">"ANNALS OF APPLIED STATISTICS"</span>,</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>            <span class="st">"PREPRINT.*"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>            <span class="st">"BIONFORMATICS JOURNAL"</span> <span class="ot">=</span> <span class="st">"BIOINFORMATICS"</span>,</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>            <span class="st">"OXFORD, ENGLAND"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>            <span class="st">"ACCEPTED"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>            <span class="st">"BMC SYST BIOL"</span> <span class="ot">=</span> <span class="st">"BMC SYSTEMS BIOLOGY"</span>,</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>            <span class="st">"COMPUT METHODS PROGRAMS BIOMED"</span> <span class="ot">=</span> <span class="st">"COMPUTER METHODS AND PROGRAMS IN BIOMEDICINE"</span>,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>            <span class="st">"CYTOMETRY A"</span> <span class="ot">=</span> <span class="st">"CYTOMETRY PART A"</span>,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>            <span class="st">"EPIGENETICS CHROMATIN"</span> <span class="ot">=</span> <span class="st">"EPIGENETICS &amp; CHROMATIN"</span>,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>            <span class="st">"F1000.*"</span> <span class="ot">=</span> <span class="st">"F1000RESEARCH"</span>,</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>            <span class="st">"FRONT BIOL"</span> <span class="ot">=</span> <span class="st">"FRONTIERS IN BIOLOGY"</span>,</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>            <span class="st">"GENOME BIOL$"</span> <span class="ot">=</span> <span class="st">"GENOME BIOLOGY"</span>,</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>            <span class="st">"GENOME RES$"</span> <span class="ot">=</span> <span class="st">"GENOME RESEARCH"</span>,</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>            <span class="st">", CODE SNIPPETS"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>            <span class="st">", SERIES B"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>            <span class="st">"J MACH LEARN RES"</span> <span class="ot">=</span> <span class="st">"JOURNAL OF MACHINE LEARNING RESEARCH"</span>,</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>            <span class="st">"METHODS MOL BIO"</span> <span class="ot">=</span> <span class="st">"METHODS IN MOLECULAR BIOLOGY"</span>,</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>            <span class="st">"MOL SYST BIOL"</span> <span class="ot">=</span> <span class="st">"MOLECULAR SYSTEMS BIOLOGY"</span>,</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NAT BIOTECH.*"</span> <span class="ot">=</span> <span class="st">"NATURE BIOTECHNOLOGY"</span>,</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NAT COMM.*"</span> <span class="ot">=</span> <span class="st">"NATURE COMMUNICATIONS"</span>,</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NAT GENET"</span> <span class="ot">=</span> <span class="st">"NATURE GENETICS"</span>,</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NAT IMMUNOL"</span> <span class="ot">=</span> <span class="st">"NATURE IMMUNOLOGY"</span>,</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NAT METH"</span> <span class="ot">=</span> <span class="st">"NATURE METHODS"</span>,</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NPG SYST BIOL APPL"</span> <span class="ot">=</span> <span class="st">"NPG SYSTEMS BIOLOGY AND APPLICATIONS"</span>,</span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>            <span class="st">" GKV873"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NUCL ACIDS RES$"</span> <span class="ot">=</span> <span class="st">"NUCLEIC ACIDS RESEARCH"</span>,</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NUCLEIC ACIDS RES$"</span> <span class="ot">=</span> <span class="st">"NUCLEIC ACIDS RESEARCH"</span>,</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>            <span class="st">"DATABASE ISSUE"</span> <span class="ot">=</span> <span class="st">""</span>,</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>            <span class="st">"OXFORD BIOINFORMATICS"</span> <span class="ot">=</span> <span class="st">"BIOINFORMATICS"</span>,</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>            <span class="st">"PLOS COMPUT BIOL"</span> <span class="ot">=</span> <span class="st">"PLOS COMPUTATIONAL BIOLOGY"</span>,</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>            <span class="st">"PLOS COMPUTAT BIOL"</span> <span class="ot">=</span> <span class="st">"PLOS COMPUTATIONAL BIOLOGY"</span>,</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>            <span class="st">"PROC NATL ACAD SCI.*"</span> <span class="ot">=</span> <span class="st">"PNAS"</span>,</span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>            <span class="st">"PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES.*"</span> <span class="ot">=</span> <span class="st">"PNAS"</span>,</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>            <span class="st">"STAT APPL GENET MOL BIOL"</span> <span class="ot">=</span> <span class="st">"STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY"</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">journal =</span> <span class="fu">str_squish</span>(journal)</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Taking a look at the first rows</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(journal_df_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             journal           package
1     BIOINFORMATICS        cytomapper
2 SCIENTIFIC REPORTS      IsoCorrectoR
3     BIOINFORMATICS             Rtpca
4     BIOINFORMATICS transcriptogramer
5     BIOINFORMATICS               ACE
6     BIOINFORMATICS          limmaGUI</code></pre>
</div>
</div>
<p>The final data frame of packages and journals where they published their papers can be explored below:</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-60789eea7ab2235f141f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-60789eea7ab2235f141f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675","676","677","678","679","680","681","682","683","684","685","686","687","688","689","690","691","692","693","694","695","696","697","698","699","700","701","702","703","704","705","706","707","708","709","710","711","712","713","714","715","716","717","718","719","720","721","722","723","724","725","726","727","728","729","730","731","732","733","734","735","736","737","738","739","740","741","742","743","744","745","746","747","748","749","750","751","752","753","754","755","756","757","758","759","760","761","762","763","764","765","766","767","768","769","770","771","772","773","774","775","776","777","778","779","780","781","782","783","784","785","786","787","788","789","790","791","792","793","794","795","796","797","798","799","800","801","802","803","804","805","806","807","808","809","810","811","812","813","814","815","816","817","818","819","820","821","822","823","824","825","826","827","828","829","830","831","832","833","834","835","836","837","838","839","840","841","842","843","844","845","846","847","848","849","850","851","852","853","854","855","856","857","858","859","860","861","862","863","864","865","866","867","868","869","870","871","872","873","874","875","876","877","878","879","880","881","882","883","884","885","886","887","888","889","890","891","892","893","894","895","896","897","898","899","900","901","902","903","904","905","906","907","908","909","910","911","912","913","914","915","916","917","918","919","920","921","922","923","924","925","926","927","928","929","930","931","932","933","934","935","936","937","938","939","940","941","942","943","944","945","946","947","948","949","950","951","952","953","954","955","956","957","958","959","960","961","962","963","964","965","966","967","968","969","970","971","972","973","974","975","976","977","978","979","980","981","982","983","984","985","986","987","988","989","990","991","992","993","994","995","996","997","998","999","1000","1001","1002","1003","1004","1005","1006","1007","1008","1009","1010","1011","1012","1013","1014","1015","1016","1017","1018","1019","1020","1021","1022","1023","1024","1025","1026","1027","1028","1029"],["BIOINFORMATICS","SCIENTIFIC REPORTS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","NATURE METHODS","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOINFORMATICS","BMC GENOMICS","PLOS ONE","NEW PHYTOLOGIST","ANALYTICAL CHEMISTRY","GENOME BIOLOGY","METHODS","GENETICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","FRONTIERS IN GENETICS","RNA BIOLOGY","BIOINFORMATICS","APPLICATIONS IN PLANT SCIENCES","PLOS ONE","GENOME BIOLOGY","BMC BIOINFORMATICS","MOLECULAR SYSTEMS BIOLOGY","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","MOLECULAR CELL","BIOINFORMATICS","COMPUT STRUCT BIOTECHNOL J","BIOINFORMATICS","F1000RESEARCH","F1000RESEARCH","BIOINFORMATICS","JOURNAL OF THE INTERNATIONAL BIOMETRIC SOCIETY","GENOME BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","GENOME BIOLOGY","SOURCE CODE FOR BIOLOGY AND MEDICINE","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","ANALYTICAL CHEMISTRY","BIOINFORMATICS","BMC GENOMICS","NATURE COMMUNICATIONS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","MOL CANCER RES","BIOINFORMATICS","BIOINFORMATICS","BMC GENOMICS","PEERJ","BIOINFORMATICS","PLOS ONE","F1000RESEARCH","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","F1000RESEARCH","NUCLEIC ACIDS RESEARCH","HAL","BIOSTATISTICS","PLOS ONE","GENOME BIOLOGY","BIOINFORMATICS","SOURCE CODE BIOL MED","F1000RESEARCH","BIOINFORMATICS","BIOINFORMATICS","NATURE COMMUNICATIONS","JOURNAL OF PROTEOME RESEARCH","BIOINFORMATICS","GENOMICS, PROTEOMICS &amp; BIOINFORMATICS","BIOINFORMATICS","NATURE COMMUNICATIONS","BIOINFORMATICS","BMC BIOINFORMATICS","NATURE COMMUNICATIONS","GENOME BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","NATURE PROTOCOLS","BIOINFORMATICS","DATABASE","BIOINFORMATICS","BMC GENOMICS","BIOINFORMATICS","GENOME BIOLOGY","PEERJ","GENOME BIOLOGY","PLOS ONE","GENOME BIOLOGY AND EVOLUTION","BMC BIOINFORMATICS","BIOINFORMATICS","COMPUTER METHODS AND PROGRAMS IN BIOMEDICINE","BMC BIOINFORMATICS","NATURE COMMUNICATIONS","BIOINFORMATICS","BIOINFORMATICS","METABOLITES","BIOINFORMATICS","PLOS ONE","BIOINFORMATICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","GENOME BIOLOGY","BRIEFINGS IN BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","101038/NMETH3869","GENOME BIOLOGY","FRONTIERS IN GENETICS","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","ANALYTICAL CHEMISTRY","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","PLOS ONE","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BRIEFINGS IN BIOINFORMATICS","THE PROSTATE","NATURE GENETICS","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","FRONTIERS IN MOLECULAR BIOSCIENCES","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","F1000RESEARCH","GENOME BIOLOGY","BIOINFORMATICS","NATURE COMMUNICATIONS","BMC BIOINFORMATICS","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","FUNCTIONAL AND INTEGRATIVE GENOMICS","BIOINFORMATICS","NATURE GENETICS","NATURE COMMUNICATIONS","BIOINFORMATICS","DEVELOPMENTAL DYNAMICS","BIOINFORMATICS","BIOINFORMATICS","NATURE COMMUNICATIONS","BIOINFORMATICS","BIOINFORMATICS","BIOMED RESEARCH INTERNATIONAL","BIOMED RESEARCH INTERNATIONAL","BIOINFORMATICS","EPIGENETICS &amp; CHROMATIN","BMC BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","BMC GENOMICS","BIOINFORMATICS","BIOINFORMATICS","ISCIENCE","BMC BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","BIONFORMATICS","BMC GENOMICS","BIOINFORMATICS","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","NAR GENOMICS AND BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","JOURNAL OF STATISTICAL SOFTWARE","BMC GENOMICS","NATURE METHODS","F1000RESEARCH","GENOME RESEARCH","BMC BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","MOLECULAR &amp; CELLULAR PROTEOMICS","BIOINFORMATICS","BIOINFORMATICS","EMBNETJOURNAL","EPIGENESYS - BIOINFORMATICS PROTOCOL","BIOINFORMATICS","BMC BIOINFORMATICS","MOL CELL PROTEOMICS","BIOINFORMATICS","COMPUTER METHODS AND PROGRAMS IN BIOMEDICINE","CURRENT PROTOCOLS","EXPERT REVIEW OF PROTEOMICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","METHODS","BMC GENOMICS","BMC BIOINFORMATICS","GENOME BIOLOGY","ANNALS OF APPLIED STATISTICS","BIOINFORMATICS","BIOINFORMATICS","PLOS ONE","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BMC BIODATA MINING","JOURNAL OF STATISTICAL PLANNING AND INFERERENCE","NATURE BIOTECHNOLOGY","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","NATURE METHODS","BIOINFORMATICS","BIOINFORMATICS","LIFE SCIENCE ALLIANCE","BIOINFORMATICS","BIOINFORMATICS","SCIENTIFIC REPORTS","F1000RESEARCH","BMC BIOINFORMATICS","PLOS ONE","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","THE JOURNAL OF OPEN SOURCE SOFTWARE","QG14","BMC BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","BMC BIOINFORMATICS","BIOINFORMATICS ADVANCES","BIOINFORMATICS","PLOS ONE","NUCLEIC ACIDS RESEARCH","GENOME BIOLOGY","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOINFORMATICS","JOURNAL OF PROTEOME RESEARCH","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","SOURCE CODE FOR BIOLOGY AND MEDICINE","ARXIV","GENOME BIOLOGY","BMC BIOINFORMATICS","BIOINFORMATICS","METABOLITES","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","PNAS","BMC GENOMICS","BMC BIOINFORMATICS","BMC GENOMICS","BMC GENOMICS","GENOME BIOLOGY","BIOINFORMATICS","ONCOTARGET","BIOINFORMATICS","101101/221499","NUCLEIC ACIDS RESEARCH","FRONTIERS IN GENETICS","BMC BIOINFORMATICS","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","GENOME BIOLOGY","ANALYTICAL CHEMISTRY","PLOS ONE","BMC BIOINFORMATICS","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","PLOS ONE","JOURNAL OF PROTEOME RESEARCH","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","NATURE METHODS","NAR GENOMICS AND BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","FRONTIERS IN GENETICS","GENOME BIOLOGY","MATHEMATICS","ANALYTICAL CHEMISTRY","BIOINFORMATICS","NATURE GENETICS","PLOS GENETICS","GENOME BIOLOGY","BIOINFORMATICS","JOURNAL OF OPEN SOURCE SOFTWARE","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS JOURNAL","PLOS ONE","BMC BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS AND BIOMEDICAL ENGINEERING","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BRIEFINGS IN BIOINFORMATICS","NAR GENOMICS AND BIOINFORMATICS","BRIEFINGS IN BIOINFORMATICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","ANALYTICAL CHEMISTRY","NATURE METHODS","NATURE METHODSODS","BIOINFORMATICS","NATURE METHODS","BMC BIOINFORMATICS","NATURE BIOTECHNOLOGY","NUCLEIC ACIDS RESEARCH","BMC BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","INT J MOL SCI","BMC BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","PLOS ONE","GENOME RESEARCH","BRIEFINGS IN BIOINFORMATICS","JOURNAL OF APPLIED STATISTICS","INTERNATIONAL JOURNAL OF EPIDEMIOLOGY","BIOINFORMATICS","BMC BIOINFORMATICS","CURRENT PROTOCOLS","BIOINFORMATICS","METHODS ENZYMOL","THE PHARMACOGENOMICS JOURNAL","METHODS IN MOLECULAR BIOLOGY","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","GENOME BIOLOGY","F1000RESEARCH","BMC BIOINFORMATICS","CURRENT PROTOCOLS","BIOSTATISTICS","ANNALS OF APPLIED STATISTICS","THE ANNALS OF APPLIED STATISTICS","BIOINFORMATICS","BIOINFORMATICS","METABOLITES","NATURE METHODS","F1000RESEARCH","BMC BIOINFORMATICS","BMC BIOINFORMATICS","JOURNAL OF BIOINFORMATICS AND COMPUTATIONAL BIOLOGY","BIOINFORMATION","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BMC GENOMICS","BIOINFORMATICS","PNAS","BMC BIOINFORMATICS","JOURNAL OF PROTEOME RESEARCH","JOURNAL OF PROTEOME RESEARCH","THE R JOURNAL","ELIFE","BIOINFORMATICS","BIOINFORMATICS","MOLECULAR AND CELLULAR PROTEOMICS","JOURNAL OF PROTEOME RESEARCH","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","ONCOTARGET","COMPUTERS IN BIOLOGY AND MEDICINE","NUCLEIC ACIDS RESEARCH","GENOME BIOLOGY","NATURE GENETICS","NATURE METHODS","NATURE METHODS","GENOME BIOLOGY","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","BMC BIOINFORMATICS","JOURNAL OF MASS SPECTROMETRY","J CHROM B","BIOINFORMATICS","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","PLOS ONE","DATABASE","JCO CLINICAL CANCER INFORMATICS","NATURE PROTOCOLS","BIOINFORMATICS,","NATURE METHODS","BMC GENOMICS","GENES","CLINICAL CANCER RESEARCH","BIOINFORMATICS","BMC BIOINFORMATICS","MOLECULAR SYSTEMS BIOLOGY","GENES &amp; GENOMICS","BMC GENOMICS","2022--2032","NATURE METHODS","PNAS","BIOINFORMATICS","NATURE BIOTECHNOLOGY","BMC BIOINFORMATICS","CANCER RESEARCH","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","METABOLITES","BIOINFORMATICS","F1000RESEARCH","GENOME BIOLOGY","BMC INFECTIOUS DISEASES","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","NATURE COMMUNICATIONS","BIOINFORMATICS","BIOINFORMATICS","AMERICAN JOURNAL OF HUMAN GENETICS","GENOME BIOLOGY","NATURE COMMUNICATIONS","NATURE COMMUNICATIONS","GENOME RESEARCH","BIOINFORMATICS","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","CYTOMETRY PART A","NATURE COMMUNICATIONS","GENOME RESEARCH","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","PATTERNS","BMC GENOMICS","ANALYTICAL CHEMISTRY","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BMC BIOINFORMATICS","NATURE COMMUNICATIONS","GENOMICS","PNAS","JOURNAL OF STATISTICAL SOFTWARE","NATURE","METABOLITES","NATURE COMMUNICATIONS","F1000RESEARCH","BIOINFORMATICS","THE AMERICAN JOURNAL OF HUMAN GENETICS","BIOINFORMATICS","PLOS ONE","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","JOURNAL OF PROTEOME RESEARCH","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","IN SILICO PLANTS","NATURE METHODS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","GENOME RESEARCH","GENOME RESEARCH","THE INNOVATION","BIOINFORMATICS","IMETA","NATURE METHODS","JOURNAL OF BIOLOGICAL RHYTHMS","BIOINFORMATICS","JOURNAL OF CLINICAL BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","NATURE COMMUNICATIONS","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","NATURE GENETICS","MOLECULAR &amp; CELLULAR PROTEOMICS","PLOS COMPUTATIONAL BIOLOGY","NUCLEIC ACIDS RESEARCH","F1000RESEARCH","JOURNAL OF ALZHEIMERS DISEASE","PLOS COMPUTATIONAL BIOLOGY","GENOME BIOLOGY","SYSTEMS BIOMEDICINE","BIOINFORMATICS","F1000RESEARCH","NUCLEIC ACIDS RESEARCH","PLOS COMPUTATIONAL BIOLOGY","BIOSTATISTICS","BIOINFORMATICS","BIOINFORMATICS","PLOS ONE","BMC BIOINFORMATICS","NATURE COMMUNICATIONS","BMC BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","NATURE COMMUNICATIONS","NATURE COMMUNICATIONS","MICROPUBLICATION BIOLOGY","BIOINFORMATICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS ADVANCES","BIOCHEMICAL AND BIOPHYSICAL RESEARCH COMMUNICATIONS","BIOINFORMATICS","GENOME BIOLOGY","NATURE COMMUNICATIONS","CELL REPORTS METHODS","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOSTATISTICS","NUCLEIC ACIDS RESEARCH","BMC BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","JOURNAL OF OPEN SOURCE SOFTWARE","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","BMC BIOINFORMATICS","NATURE GENETICS","PLOS COMPUTATIONAL BIOLOGY","FRONTIERS IN GENETICS","1: 413 BYTES SKIPPEDQ, SCOTT-BOYER, MP, TAYEB, G, IMHOLTE, LABBE, A, DESCHEPPER C, AND GOTTARDO R, APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY, 2012, 11, 2211 BYTES SKIPPED","METABOLITES","NUCLEIC ACIDS RESEARCH","JOURNAL OF STATISTICAL SOFTWARE","BMC BIOINFORMATICS","BIOINFORMATICS","NATURE GENETICS","BIOINFORMATICS","CELL REPORTS","NATURE METHODS","BMC BIOINFORMATICS","JOURNAL OF MOLECULAR BIOLOGY","BIOINFORMATICS","FRONTIERS IN ONCOLOGY","ANALYTICAL CHEMISTRY","BMC BIOINFORMATICS","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","NATURE BIOTECHNOLOGY","PLOS COMPUTATIONAL BIOLOGY","GENOME BIOLOGY","BIOINFORMATICS","BIOSTATISTICS","GENOME RESEARCH","PNAS","F1000RESEARCH","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","NATURE METHODS","GENOME BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","JOURNAL OF MACHINE LEARNING RESEARCH","J COMPUT BIOL","GENETICS","J R STAT SOC SER C-APPL STAT","BIOINFORMATICS","CYTOMETRY PART A","JOURNAL OF COMPUTATIONAL SCIENCE","JOURNAL OF THE ROYAL STATISTICAL SOCIETY","BIOINFORMATICS","GENOME BIOLOGY","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOMETRICS","DOI: 101111/BIOM12965","PLOS COMPUTATIONAL BIOLOGY","NATURE METHODS","BIOINFORMATICS","HUMAN MUTATION","PLANT METHODS","BIOINFORMATICS","F1000RESEARCH","BMC BIOINFORMATICS","GIGASCIENCE","BIOINFORMATICS","BIOINFORMATICS","METABOLOMICS","BMC BIOINFORMATICS","BIOINFORMATICS","GENOME RESEARCH","JOURNAL OF THE NATIONAL CANCER INSTITUTE","NUCLEIC ACIDS RESEARCH","MICROBIOME","GENOME RESEARCH","BIOINFORMATICS","BMC GENOMICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","CELL REPORTS","STAR PROTOCOLS","NATURE COMMUNICATIONS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","SCIENTIFIC REPORTS","NATURE IMMUNOLOGY","GENOME BIOLOGY","BIOCHIMICA ET BIOPHYSICA ACTA BBA - BIOENERGETICS","METHODS INF MED","BIOINFORMATICS","PEERJ","NATURE COMMUNICATIONS","NATURE METHODS","BIOINFORMATICS","CELL","BMC BIOINFORMATICS","MOLECULAR SYSTEMS BIOLOGY","F1000RESEARCH","NATURE BIOTECHNOLOGY","NATURE METHODS","BMC GENOMICS","BMC GENOMICS","BMC GENOMICS","BMC GENOMICS","BIOINFORMATICS","METHODS IN ECOLOGY AND EVOLUTION","MOLECULAR &amp; CELLULAR PROTEOMICS","MOLECULAR &amp; CELLULAR PROTEOMICS","ANALYTICAL CHEMISTRY","CELL","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","CYTOMETRY PART A","GENOME BIOLOGY","PLOS ONE","NATURE METHODS","BIOINFORMATICS","GENOMICS, PROTEOMICS &amp; BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","MOLECULAR &amp; CELLULAR PROTEOMICS","BIOINFORMATICS","PLOS ONE","FRONTIERS IN BIOLOGY","BIOMETRICS","CANCER INFORMATICS","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","BMC BIOINFORMATICS","SAGMB","BMC BIOINFORMATICS","J CHEM INF MODEL","CYTOMETRY PART A","EPIGENETICS &amp; CHROMATIN","NUCLEIC ACIDS RESEARCH","NATURE BIOTECHNOLOGY","BIOINFORMATICS","NPJ SYSTEMS BIOLOGY AND APPLICATIONS","BMC BIOINFORMATICS","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","GENOME BIOLOGY","CELL SYSTEMS","BIOINFORMATICS","PLOS ONE","BMC BIOINFORMATICS","ANNALS OF APPLIED STATISTICS","WELLCOME OPEN RESEARCH","BIOINFORMATICS","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOSTATISTICS","JOURNAL OF STATISTICAL SOFTWARE","ANNALS OF APPLIED STATISTICS","BIOINFORMATICS","F1000RESEARCH","BIOINFORMATICS","BIOINFORMATICS","JOURNAL OF PROTEOME RESEARCH","GENOME RESEARCH","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","BIOSTATISTICS","BMC BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","ANALYTICA CHIMICA ACTA","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","F1000RESEARCH","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","NATURE METHODS","JOURNAL OF MACHINE LEARNING RESEARCH","BIOINFORMATICS","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","NATURE METHODS","*** PLOS ONE, 2013, VOLUME 8, ISSUE 7, E67019","*** MICROBIOME, 2014, VOLUME 2, 15","*** JOURNAL OF COMPUTATIONAL AND GRAPHICAL STATISTICS,","F1000RESEARCH","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","NATURE PROTOCOLS","BIOINFORMATICS","BIOINFORMATICS","BMC MEDICAL GENOMICS","GENOME BIOLOGY","NATURE METHODS","BRIEFINGS IN BIOINFORMATICS","BIOINFORMATICS","NATURE METHODS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","GENOME RESEARCH","GENOME BIOLOGY","GENOME BIOLOGY","BMC BIOINFORMATICS","CELL REPORTS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","OMICS: A JOURNAL OF INTEGRATIVE BIOLOGY","NATURE METHODS","PLANTS","NATURE GENETICS","BIOINFORMATICS","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","FRONTIERS IN GENETICS","GENOME RESEARCH","BMC BIOINFORMATICS","PEERJ","FRONTIERS IN NEUROSCIENCE","NUCLEIC ACIDS RESEARCH","GENOME BIOLOGY","NATURE METHODS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","MOLECULAR BIOLOGY AND EVOLUTION","BMC BIOINFORMATICS","ALGORITHMS FOR MOLECULAR BIOLOGY","GENOME BIOLOGY","ALGORITHMS FOR MOLECULAR BIOLOGYL","BMC BIOINFORMATICS","PHYSIOL GENOMICS","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BMC BIOINFORMATICS","JAPANESE J APPL STAT","BIOSTATISTICS","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","PLOS COMPUTATIONAL BIOLOGY","JOURNAL OF STATISTICAL SOFTWARE","BMC BIOINFORMATICS","MOLECULAR SYSTEMS BIOLOGY","JASMS","ANALYTICAL CHEMISTRY","NATURE METHODS","EPIGENOMICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","NATURE METHODS","EPIGENOMICS","BIOINFORMATICS","EPIGENETICS &amp; CHROMATIN","NATURE COMMUNICATIONS","ANALYTICA CHIMICA ACTA","BIOMETRICS","BMC BIOINFORMATICS","MOLECULAR SYSTEMS BIOLOGY","F1000RESEARCH","BIOINFORMATICS","BMC INFORMATICS","BMC BIOINFORMATICS","GENOME BIOLOGY","BMC CANCER","BMC BIOINFORMATICS","BMC EPIGENETICS &amp; CHROMATIN","AGING","BMC GENOMICS","BMC BIOINFORMATICS","ANIMAL GENETICS","ANIMAL GENETICS","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","F1000RESEARCH","BIOINFORMATICS","BMC GENOMICS","JOURNAL OF CACHEXIA, SARCOPENIA AND MUSCLE","JOURNAL OF IMMUNOLOGICAL METHODS","SCIENTIFIC REPORTS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BIOSTATISTICS","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOINFORMATICS","NATURE METHODS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","THE ANNALS OF APPLIED STATISTICS","BIOINFORMATICS","J PROTEOMICS","MOL CELL PROTEOMICS","PLOS COMPUTATIONAL BIOLOGY","F1000RESEARCH","F1000RESEARCH","MOLECULAR BIOLOGY AND EVOLUTION","BMC BIOINFORMATICS","PLOS ONE","BMC SYSTEMS BIOLOGY","BMC BIOINFORMATICS","BMC BIOINFORMATICS","CURRENT PROTOCOLS","PLOS ONE","MOLECULAR BIOSYSTEMS","NUCLEIC ACIDS RESEARCH","BRIEFINGS IN BIOINFORMATICS","NATURE COMMUNICATIONS","BMC BIOINFORMATICS","PEERJ","NUCLEIC ACIDS RESEARCH","CLINICAL EPIGENETICS","BIOINFORMATICS","BIOINFORMATICS","BMC GENOMICS","EPIGENETICS","BIOINFORMATICS","PLOS COMPUTATIONAL BIOLOGY","BMC BIOINFORMATICS","METHODS IN MOLECULAR BIOLOGY","BIOINFORMATICS","CLINICAL CANCER RESEARCH","BIOINFORMATICS","THE JOURNAL OF OPEN SOURCE SOFTWARE","IEEE/ACM TRANS COMPUT BIOL BIOINFORM","BIOINFORMATICS","GENOME BIOLOGY","BMC GENOMICS","PNAS","ANALYTICAL CHEMISTRY","NATURE COMMUNICATIONS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BIOINFORMATICS","CURRENT PROTOCOLS","BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","CHROMOSOME RESEARCH","BMC BIOINFORMATICS","BRIEFINGS IN BIOINFORMATICS","THE INNOVATION","OMICS: A JOURNAL OF INTEGRATIVE BIOLOGY","BIOMETRICS","STATISTICA SINICA","BIOINFORMATICS","BIOINFORMATICS","SCIENTIFIC REPORTS","BIOINFORMATICS","BIOINFORMATICS","BMC SYSTEMS BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","ALGORITHMICA","SCIENTIFIC REPORTS","NUCLEIC ACIDS RESEARCH","MOLECULAR SYSTEMS BIOLOGY","NATURE METHODS","NUCLEIC ACIDS RESEARCH","TCBB/IEEE","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","BIOLOGY DIRECT","JASA","NPJ SYST BIOL APPL","BIOINFORMATICS","GENOME BIOLOGY","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","GENOME BIOLOGY","EPIGENETICS &amp; CHROMATIN","BIOINFORMATICS","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY","METHODS IN MOLECULAR BIOLOGY","BIOINFORMATICS","METHODS IN MOLECULAR BIOLOGYL","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","EPIGENETICS &amp; CHROMATIN","BMC BIOINFORMATICS","BIOINFORMATICS","BIOINFORMATICS","BMC BIOINFORMATICS","BMC BIOINFORMATICS","METHODS IN ECOLOGY AND EVOLUTION","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","MOLECULAR SYSTEMS BIOLOGY","GENOME BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","NATURE BIOTECHNOLOGY","F1000RESEARCH","NUCLEIC ACIDS RESEARCH","BMC BIOINFORMATICS","GENOME RESEARCH","BIOINFORMATICS","BIOSTATISTICS","NUCLEIC ACIDS RESEARCH","BMC BIOINFORMATICS","BIOINFORMATICS","BMC SYSTEMS BIOLOGY","TECHNOMETRICS","INT J MOL SCI","GENOME BIOLOGY","PNAS","BIOINFORMATICS","PNAS","BIOINFORMATICS","GENOME BIOLOGY","NUCLEIC ACIDS RESEARCH","NATURE METHODS","NUCLEIC ACIDS RESEARCH","BIOINFORMATICS","BIOINFORMATICS","GENOME BIOLOGY","BIOINFORMATICS","BIOINFORMATICS","IMETA","CURRENT PROTOCOLS IN BIOINFORMATICS","MOLECULAR BIOLOGY AND EVOLUTION","METHODS IN ECOLOGY AND EVOLUTION","BMC GENOMICS","BRIEFINGS IN BIOINFORMATICS","CELL GENOMICS","NUCLEIC ACIDS RESEARCH","NUCLEIC ACIDS RESEARCH","SCIENCE","BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","NATURE BIOTECHNOLOGY","MOL CELL PROTEOMICS","NATURE BIOTECHNOLOGY","MOLECULAR SYSTEMS BIOLOGY","BIOINFORMATICS"],["cytomapper","IsoCorrectoR","Rtpca","transcriptogramer","ACE","limmaGUI","DMRforPairs","iCOBRA","iGC","affylmGUI","npGSEA","ILoReg","R4RNA","compcodeR","farms","ABSSeq","phyloseq","VaSP","CAMERA","RedeR","BindingSiteFinder","GBScleanR","ggcyto","hapFabia","ChIPQC","miRSM","randRotation","flowPloidy","pgca","BaalChIP","ChIPseqR","dyebias","consensusSeekeR","cicero","rBiopaxParser","UCell","RareVariantVis","shinyMethyl","GOsummaries","BLMA","SOMNiBUS","RTopper","TFEA.ChIP","MetCirc","OLINgui","OLINgui","PathNet","Path2PPI","GSAR","DEsubs","methInheritSim","MetNet","GOstats","geNetClassifier","crisprViz","PAA","IPO","APAlyzer","bacon","IsoformSwitchAnalyzeR","IsoformSwitchAnalyzeR","BPRMeth","mitch","RITAN","TargetScore","rfaRm","microbiomeDASim","CHETAH","motifbreakR","cellHTS2","ELMER","scDblFinder","idr2d","epistack","padma","miRLAB","DESeq2","GENESIS","goSTAG","seqCAT","CTSV","ASpli","MetaNeighbor","rawrr","dittoSeq","synergyfinder","nondetects","crisprBwa","crisprBwa","Risa","crisprBowtie","crisprBowtie","massiR","ramr","methylKit","ensembldb","DEP","waddR","fobitools","TimiRGeN","FindIT2","HiCBricks","scCB2","HiLDA","quantro","dagLogo","sangeranalyseR","tripr","annotatr","Director","diffUTR","SpotClean","HiTC","LedPred","Spectra","fCCAC","prebs","ChIPComp","AllelicImbalance","LowMACA","DMRcaller","slalom","metaseqR2","benchdamic","benchdamic","PROPER","dada2","MuData","normalize450K","speckle","ReactomeGraph4R","peakPantheR","mistyR","monaLisa","msPurity","psichomics","AssessORF","SGSeq","anota2seq","POWSC","microSTASIS","matchBox","matchBox","SANTA","fabia","shinyepico","biosigner","UMI4Cats","chipenrich","CNVMetrics","methylCC","SplicingFactory","TPP2D","multiGSEA","ngsReports","iSEE","AlpsNMR","tximport","Dino","BioNERO","customProDB","RTN","RTN","GeneRegionScan","sangerseqR","IgGeneUsage","FamAgg","tradeSeq","InterMineR","twoddpcr","biomvRCNS","Guitar","DEsingle","DAMEfinder","NBAMSeq","NoRCE","CNVRanger","MutationalPatterns","R453Plus1Toolbox","CHRONOS","DeMixT","SRAdb","drawProteins","seq2pathway","switchBox","switchBox","ProteoDisco","GOSim","CAGEr","ppcseq","macat","altcdfenvs","GeneGeneInteR","ReadqPCR","Anaquin","ExploreModelMatrix","pram","GEM","GEM","pipeComp","NPARC","MIRA","MWASTools","CexoR","CexoR","mnem","EDASeq","procoil","HilbertCurve","MLSeq","scp","scp","countsimQC","ChAMP","ChAMP","ChAMP","slingshot","gtrellis","topconfects","VanillaICE","scanMiR","Omixer","GOTHiC","RCAS","SNPRelate","ViSEAGO","hopach","CrispRVariants","mCSEA","xmapbridge","specL","chromVAR","easyRNASeq","DaMiRseq","multicrispr","LOLA","IWTomics","corral","InteractionSet","methylPipe","BiGGR","singscore","singscore","GenomicTuples","coMET","coMET","R3CPET","ramwas","GenomicFeatures","IntEREst","decoupleR","CoRegNet","RNAAgeCalc","fishpond","ggbio","splatter","Ularcirc","ConsensusClusterPlus","HilbertVis","XINA","QuaternaryProd","QuaternaryProd","PureCN","hierinf","MPRAnalyze","CAGEfightR","microbiomeMarker","CompoundDb","HTqPCR","scater","rprimer","qsea","NetPathMiner","biocViews","cycle","KEGGgraph","BiRewire","BiRewire","pathifier","scReClassify","PCAN","NormqPCR","GenomicInteractions","scMET","sarks","Uniquorn","regioneR","decontam","REMP","categoryCompare","Ringo","plyranges","limma","ADaCGH2","tidybulk","metabCombiner","RTCGAToolbox","diffHic","clippda","idpr","MsBackendRawFileReader","pandaR","EGAD","censcyt","deco","GSCA","GenVisR","scmap","recountmethylation","POMA","GWASTools","bigmelon","epigenomix","ChIPanalyser","sigFeature","BANDITS","DExMA","yamss","RTNsurvival","RTNsurvival","packFinder","tricycle","scFeatures","basilisk","MethReg","Metab","srnadiff","variancePartition","variancePartition","scFeatureFilter","Nebulosa","miRNAmeConverter","SeqGSEA","SeqGSEA","ORFik","Linnorm","SpatialExperiment","GSEABenchmarkeR","dearseq","FEAST","DiffLogo","plgem","LOBSTAHS","BUMHMM","cydar","karyoploteR","SingleCellExperiment","IntOMICS","batchelor","qusage","qusage","qusage","MiRaGE","miRcomp","signeR","Rmmquant","timeOmics","srnadiff","FRGEpistasis","coseq","coseq","bumphunter","bumphunter","pcaExplorer","pcaExplorer","glmGamPoi","ACME","HIBAG","HIBAG","spatzie","ExperimentSubset","bsseq","scran","GeneTonic","GeneTonic","cqn","DEGraph","DEGraph","genomation","HTSFilter","MetaboAnnotation","TAPseq","clustifyr","InTAD","systemPipeR","Mfuzz","Mfuzz","made4","wpm","maskBAD","maskBAD","mpra","structToolbox","GraphAlignment","HiCcompare","synapter","synapter","DECIPHER","philr","h5vc","awst","MSstatsPTM","ropls","ReportingTools","graphite","graphite","hypeR","hipathia","KnowSeq","methylInheritance","COCOA","Pi","BiocGenerics","Biobase","scDD","nucleoSim","nucleoSim","TissueEnrich","recount3","pwrEWAS","RMassBank","metaMS","MetaboSignal","CNVfilteR","netSmooth","omicsPrint","SWATH2stats","gemma.R","cBioPortalData","biomaRt","biomaRt","trackViewer","GUIDEseq","FoldGO","ffpe","BDMMAcorrect","MeSHDbi","DominoEffect","IVAS","EnrichedHeatmap","QDNAseq","RNAinteract","scMerge","multiHiCcompare","RUVSeq","minet","MultiAssayExperiment","csaw","csaw","CellBench","MsBackendMsp","codelink","BiocWorkflowTools","Chicago","TBSignatureProfiler","PharmacoGx","tilingArray","syntenet","ChromSCape","MSnbase","MEDIPS","Summix","DropletUtils","DropletUtils","FRASER","NADfinder","copa","TurboNorm","FlowSOM","icetea","gcapc","COHCAP","TFBSTools","easier","GenomicDistributions","topdownr","apeglm","m6Aboost","SamSPECTRAL","zinbwave","SCAN.UPC","SCAN.UPC","lmdme","DiffBind","MetaboCoreUtils","GenomicSuperSignature","biobtreeR","ClusterSignificance","OUTRIDER","methylGSA","DepecheR","epiNEM","CancerSubtypes","EnMCB","struct","vsn","NormalyzerDE","mixOmics","GEOquery","cageminer","RcisTarget","esATAC","survcomp","survcomp","maftools","amplican","MicrobiotaProcess","ComplexHeatmap","ComplexHeatmap","IHW","rain","oposSOM","oppar","omicade4","EDIRquery","AffyRNADegradation","globalSeq","bnem","fastseg","spaSim","DELocal","RTNduals","RTNduals","MSstatsLOBD","tximeta","piano","satuRn","miRNApath","Maaslin2","goseq","rScudo","rScudo","BgeeDB","BgeeDB","miQC","dmrseq","VariantAnnotation","BatchQC","BEclear","gage","qPLEXanalyzer","EnrichmentBrowser","TreeSummarizedExperiment","airpart","ANCOMBC","ANCOMBC","rrvgo","affy","miRBaseConverter","miRspongeR","miRspongeR","supraHex","PhyloProfile","SIAMCAT","HiCDCPlus","iPath","OncoSimulR","OncoSimulR","OncoSimulR","OncoSimulR","girafe","nullranges","nullranges","plotgardener","ShortRead","RUVnormalize","wavClusteR","wavClusteR","atSNP","rmspc","scPCA","scPCA","scPipe","groHMM","viper","ROTS","TADCompare","iBMQ","MsCoreUtils","rqubic","Rqc","TargetSearch","SAIGEgds","SAIGEgds","SAIGEgds","transite","motifStack","GSVA","APL","methylSig","SCFA","xcms","xcms","xcms","NanoMethViz","eisaR","eisaR","SEPIRA","SEPIRA","qsmooth","DEXSeq","DEXSeq","KEGGlincs","ArrayExpress","pathview","annotationTools","PSEA","dcanr","SomaticSignatures","EGSEA","CiteFuse","qpgraph","qpgraph","qpgraph","qpgraph","beadarray","CytoML","LACE","globaltest","globaltest","missMethyl","missMethyl","missMethyl","missMethyl","DMCFB","DMCFB","beachmat","AUCell","projectR","panelcn.mops","CSAR","FGNet","DRIMSeq","Harman","getDEE2","methrix","ASICS","ASICS","frmaTools","ClassifyR","TitanCNA","doppelgangR","MethylSeekR","animalcules","MEDME","oligo","CeTF","BioNet","BioNet","destiny","PhosR","PhosR","erccdashboard","SeqArray","SeqArray","kebabs","cliqueMS","Qtlizer","SingleR","CellMapper","ComPrAn","GlobalAncova","GlobalAncova","ceRNAnetsim","progeny","transformGamPoi","paxtoolsr","proActiv","GARS","cosmosR","RCy3","monocle","monocle","wateRmelon","EventPointer","EventPointer","ATACseqQC","meshes","miaSim","msqrob2","msqrob2","msqrob2","diggit","diggit","systemPipeShiny","EBImage","OmaDB","SummarizedBenchmark","dce","SIM","flowClust","flowClust","Melissa","GENIE3","GENIE3","cleanUpdTSeq","simplifyEnrichment","derfinderPlot","RImmPort","SPONGE","Rcpi","MSstatsTMT","Prostar","CRISPRseek","CRISPRseek","PICS","BeadDataPackR","RJMCMCNucleosomes","tweeDEseq","DMRScan","rgsepd","bioassayR","flowClean","DMRcate","DMRcate","densvis","onlineFDR","sRACIPE","HPAanalyze","rcellminer","BASiCS","BASiCS","BASiCS","triplex","messina","Trendy","casper","mfa","rGREAT","MassSpecWavelet","iSEEhex","Cardinal","crlmm","crlmm","crlmm","crlmm","gaga","regutools","illuminaio","RankProd","RankProd","isobar","NOISeq","NOISeq","derfinder","derfinder","MultiDataSet","CoreGx","netZooR","PepsNMR","rtracklayer","TCGAbiolinks","TCGAbiolinks","TCGAbiolinks","TCGAbiolinks","TRESS","snifter","snifter","QuasR","QuasR","QuasR","QuasR","ALDEx2","ALDEx2","ALDEx2","rDGIdb","rDGIdb","DOSE","TEQC","rTRM","signatureSearch","octad","pcaMethods","Glimma","Pigengene","rnaseqcomp","SCnorm","ISAnalytics","GenomicScores","cytoMEM","nempi","flowAI","Repitools","Repitools","Repitools","gscreend","dupRadar","CellTrails","netresponse","regionReport","GeneExpressionSignature","cardelino","PAST","gcatest","gcatest","MAIT","Clomial","RandomWalkRestartMH","derfinderHelper","INSPEcT","INSPEcT","INSPEcT","CEMiTool","MouseFM","EWCE","cn.mops","RnBeads","RnBeads","flowPeaks","DiscoRhythm","arrayQualityMetrics","ggtreeExtra","TCC","TCC","TCC","TCC","TCC","TCC","TCC","TCC","TCC","TCC","TCC","TCC","TCC","CNEr","rbsurv","csdR","scClassify","BRAIN","BRAIN","EpiDISH","EpiDISH","EpiDISH","EpiDISH","EpiDISH","EpiDISH","snm","MAGAR","SPIAT","statTarget","DMCHMM","puma","netDx","netDx","unifiedWMWqPCR","ccImpute","DART","DART","DART","DART","planet","planet","planet","meshr","meshr","meshr","edgeR","edgeR","edgeR","MSstats","methimpute","MEAT","openPrimeR","openPrimeR","pqsfinder","funtooNorm","InteractiveComplexHeatmap","DSS","DSS","DSS","DSS","MethylAid","SC3","diffuStats","diffuStats","metaCCA","fmrs","pRoloc","pRoloc","pRoloc","pRoloc","pRoloc","pRoloc","treeio","FELLA","FELLA","CellNOptR","SpectralTAD","ideal","ideal","PING","ReactomePA","cola","cola","MoonlightR","decompTumor2Sig","decompTumor2Sig","ENmix","ENmix","ENmix","ENmix","ENmix","ENmix","ENmix","openCyto","ChIPpeakAnno","ChIPpeakAnno","GEOmetadb","consensusOV","lfa","biotmle","netboost","RadioGx","AneuFinder","BioQC","BioQC","rgoslin","deepSNV","deepSNV","DAPAR","bcSeq","metabinR","DNABarcodeCompatibility","ChIPseeker","ChIPseeker","gdsfmt","gdsfmt","chromDraw","CytoGLMM","ggmsa","clusterProfiler","clusterProfiler","HDTD","HDTD","HDTD","DeepBlueR","runibic","runibic","consensus","StarBioTrek","Rdisop","Rdisop","Rdisop","lineagespot","Rsubread","OmnipathR","OmnipathR","RPA","RPA","DNABarcodes","STRINGdb","lumi","lumi","lumi","lumi","XDE","CARNIVAL","minfi","minfi","minfi","minfi","minfi","minfi","minfi","ternarynet","GOSemSim","GOSemSim","GOSemSim","DNAshapeR","DNAshapeR","DNAshapeR","DNAshapeR","aroma.light","aroma.light","aroma.light","aroma.light","aroma.light","LEA","HilbertVisGUI","cn.farms","MOFA2","MOFA2","msa","QUBIC","recount","recount","recount","recount","recount","methylclock","frma","frma","frma","frma","CNORode","affyPLM","SpidermiR","vulcan","mosbi","tigre","tigre","tigre","Rbowtie","Rbowtie","metagenomeSeq","multiMiR","seqbias","TOAST","TOAST","TOAST","TOAST","ggtree","ggtree","ggtree","ggtree","debrowser","sesame","sesame","sesame","sesame","BSgenome","Rbowtie2","muscle","mzR","mzR","mzR","mzR","mzR"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>journal<\/th>\n      <th>package<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="summary-stats" class="level2">
<h2 class="anchored" data-anchor-id="summary-stats">Summary stats</h2>
<p>Now, let’s count the frequency of packages in each journal and show the top 20 journals based number of the number of papers associated with Bioc packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 journals in number of papers associated with Bioc pkgs</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>citation_stats <span class="ot">&lt;-</span> journal_df_clean <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(journal) <span class="sc">%&gt;%</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="sc">-</span>n) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>citation_stats</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                      journal   n
1                                              BIOINFORMATICS 303
2                                          BMC BIOINFORMATICS 101
3                                      NUCLEIC ACIDS RESEARCH  71
4                                              GENOME BIOLOGY  61
5                                               F1000RESEARCH  34
6                                              NATURE METHODS  30
7                                                BMC GENOMICS  25
8                                       NATURE COMMUNICATIONS  24
9                                  PLOS COMPUTATIONAL BIOLOGY  22
10                                                   PLOS ONE  22
11                                            GENOME RESEARCH  13
12                                       ANALYTICAL CHEMISTRY  11
13                                              BIOSTATISTICS  10
14                                BRIEFINGS IN BIOINFORMATICS   9
15                                       NATURE BIOTECHNOLOGY   9
16                               JOURNAL OF PROTEOME RESEARCH   8
17                                  MOLECULAR SYSTEMS BIOLOGY   8
18                                            NATURE GENETICS   8
19                                                       PNAS   8
20 STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY   7</code></pre>
</div>
</div>
<p>Exploring it visually:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read figure with Bioc logo</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>bioc_logo <span class="ot">&lt;-</span> png<span class="sc">::</span><span class="fu">readPNG</span>(</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>    here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"blog"</span>, <span class="st">"2022-01-03-bioc_publications"</span>, <span class="st">"featured-bioc.png"</span>), </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">native =</span> <span class="cn">TRUE</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Define plotting params</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>last_updated <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">Sys.Date</span>(), <span class="st">"%Y-%m-%d"</span>)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>xmax <span class="ot">&lt;-</span> <span class="fu">max</span>(citation_stats<span class="sc">$</span>n) <span class="sc">+</span> <span class="dv">30</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>xmax <span class="ot">&lt;-</span> <span class="fu">round</span>(xmax <span class="sc">/</span> <span class="dv">10</span>) <span class="sc">*</span> <span class="dv">10</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot data</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(citation_stats, <span class="fu">aes</span>(<span class="at">x =</span> n, <span class="at">y =</span> <span class="fu">reorder</span>(journal, n))) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> n), <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="dv">0</span>, xmax) <span class="sc">+</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Where are papers associated with BioC packages published?"</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"Last update: "</span>, last_updated),</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Number of papers"</span>, <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>    patchwork<span class="sc">::</span><span class="fu">inset_element</span>(</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>        bioc_logo,</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">left =</span> <span class="fl">0.5</span>,</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">top =</span> <span class="fl">0.55</span>,</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">right =</span> <span class="fl">0.95</span>,</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">bottom =</span> <span class="fl">0.3</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_void</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
<p>And voilà! In case you want to explore the whole table, here it is:</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-64b93575ba3a55052b1f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-64b93575ba3a55052b1f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"],["BIOINFORMATICS","BMC BIOINFORMATICS","NUCLEIC ACIDS RESEARCH","GENOME BIOLOGY","F1000RESEARCH","NATURE METHODS","BMC GENOMICS","NATURE COMMUNICATIONS","PLOS COMPUTATIONAL BIOLOGY","PLOS ONE","GENOME RESEARCH","ANALYTICAL CHEMISTRY","BIOSTATISTICS","BRIEFINGS IN BIOINFORMATICS","NATURE BIOTECHNOLOGY","JOURNAL OF PROTEOME RESEARCH","MOLECULAR SYSTEMS BIOLOGY","NATURE GENETICS","PNAS","STATISTICAL APPLICATIONS IN GENETICS AND MOLECULAR BIOLOGY"],[303,101,71,61,34,30,25,24,22,22,13,11,10,9,9,8,8,8,8,7]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>journal<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="session-information" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="session-information">Session information</h2>
<p>This post was created under the following conditions:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.3.0 (2023-04-21)
 os       Ubuntu 20.04.5 LTS
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Europe/Brussels
 date     2023-06-13
 pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 BiocManager   1.30.20 2023-02-24 [1] CRAN (R 4.3.0)
 BiocStyle     2.28.0  2023-04-25 [1] Bioconductor
 bslib         0.4.2   2022-12-16 [1] CRAN (R 4.3.0)
 cachem        1.0.8   2023-05-01 [1] CRAN (R 4.3.0)
 cli           3.6.1   2023-03-23 [1] CRAN (R 4.3.0)
 colorspace    2.1-0   2023-01-23 [1] CRAN (R 4.3.0)
 crosstalk     1.2.0   2021-11-04 [1] CRAN (R 4.3.0)
 curl          5.0.0   2023-01-12 [1] CRAN (R 4.3.0)
 digest        0.6.31  2022-12-11 [1] CRAN (R 4.3.0)
 dplyr       * 1.1.2   2023-04-20 [1] CRAN (R 4.3.0)
 DT            0.27    2023-01-17 [1] CRAN (R 4.3.0)
 ellipsis      0.3.2   2021-04-29 [1] CRAN (R 4.3.0)
 evaluate      0.20    2023-01-17 [1] CRAN (R 4.3.0)
 fansi         1.0.4   2023-01-22 [1] CRAN (R 4.3.0)
 farver        2.1.1   2022-07-06 [1] CRAN (R 4.3.0)
 fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.3.0)
 forcats     * 1.0.0   2023-01-29 [1] CRAN (R 4.3.0)
 generics      0.1.3   2022-07-05 [1] CRAN (R 4.3.0)
 ggplot2     * 3.4.2   2023-04-03 [1] CRAN (R 4.3.0)
 glue          1.6.2   2022-02-24 [1] CRAN (R 4.3.0)
 gtable        0.3.3   2023-03-21 [1] CRAN (R 4.3.0)
 here          1.0.1   2020-12-13 [1] CRAN (R 4.3.0)
 hms           1.1.3   2023-03-21 [1] CRAN (R 4.3.0)
 htmltools     0.5.5   2023-03-23 [1] CRAN (R 4.3.0)
 htmlwidgets   1.6.2   2023-03-17 [1] CRAN (R 4.3.0)
 httr          1.4.5   2023-02-24 [1] CRAN (R 4.3.0)
 jquerylib     0.1.4   2021-04-26 [1] CRAN (R 4.3.0)
 jsonlite      1.8.4   2022-12-06 [1] CRAN (R 4.3.0)
 knitr         1.42    2023-01-25 [1] CRAN (R 4.3.0)
 labeling      0.4.2   2020-10-20 [1] CRAN (R 4.3.0)
 lifecycle     1.0.3   2022-10-07 [1] CRAN (R 4.3.0)
 lubridate   * 1.9.2   2023-02-10 [1] CRAN (R 4.3.0)
 magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.3.0)
 munsell       0.5.0   2018-06-12 [1] CRAN (R 4.3.0)
 patchwork     1.1.2   2022-08-19 [1] CRAN (R 4.3.0)
 pillar        1.9.0   2023-03-22 [1] CRAN (R 4.3.0)
 pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.3.0)
 png           0.1-8   2022-11-29 [1] CRAN (R 4.3.0)
 purrr       * 1.0.1   2023-01-10 [1] CRAN (R 4.3.0)
 R6            2.5.1   2021-08-19 [1] CRAN (R 4.3.0)
 readr       * 2.1.4   2023-02-10 [1] CRAN (R 4.3.0)
 rlang         1.1.1   2023-04-28 [1] CRAN (R 4.3.0)
 rmarkdown     2.21    2023-03-26 [1] CRAN (R 4.3.0)
 rprojroot     2.0.3   2022-04-02 [1] CRAN (R 4.3.0)
 rstudioapi    0.14    2022-08-22 [1] CRAN (R 4.3.0)
 rvest       * 1.0.3   2022-08-19 [1] CRAN (R 4.3.0)
 sass          0.4.5   2023-01-24 [1] CRAN (R 4.3.0)
 scales        1.2.1   2022-08-20 [1] CRAN (R 4.3.0)
 sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.3.0)
 stringi       1.7.12  2023-01-11 [1] CRAN (R 4.3.0)
 stringr     * 1.5.0   2022-12-02 [1] CRAN (R 4.3.0)
 tibble      * 3.2.1   2023-03-20 [1] CRAN (R 4.3.0)
 tidyr       * 1.3.0   2023-01-24 [1] CRAN (R 4.3.0)
 tidyselect    1.2.0   2022-10-10 [1] CRAN (R 4.3.0)
 tidyverse   * 2.0.0   2023-02-22 [1] CRAN (R 4.3.0)
 timechange    0.2.0   2023-01-11 [1] CRAN (R 4.3.0)
 tzdb          0.3.0   2022-03-28 [1] CRAN (R 4.3.0)
 utf8          1.2.3   2023-01-31 [1] CRAN (R 4.3.0)
 vctrs         0.6.2   2023-04-19 [1] CRAN (R 4.3.0)
 withr         2.5.0   2022-03-03 [1] CRAN (R 4.3.0)
 xfun          0.39    2023-04-20 [1] CRAN (R 4.3.0)
 xml2          1.3.4   2023-04-27 [1] CRAN (R 4.3.0)
 yaml          2.3.7   2023-01-23 [1] CRAN (R 4.3.0)

 [1] /home/faalm/R/x86_64-pc-linux-gnu-library/4.3
 [2] /usr/local/lib/R/site-library
 [3] /usr/lib/R/site-library
 [4] /usr/lib/R/library

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/almeidasilvaf\.github\.io\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<script src="https://giscus.app/client.js" data-repo="almeidasilvaf/blogcomments" data-repo-id="R_kgDOJuYlZg" data-category="General" data-category-id="DIC_kwDOJuYlZs4CXJi8" data-mapping="title" data-reactions-enabled="1" data-emit-metadata="0" data-input-position="top" data-theme="light" data-lang="en" crossorigin="anonymous" async="">
</script>
<script type="application/javascript">
  const giscusIframeObserver = new MutationObserver(function (mutations) {
    mutations.forEach(function (mutation) {
      mutation.addedNodes.forEach(function (addedNode) {
        if (addedNode.matches && addedNode.matches('div.giscus')) {
          const giscusIframe = addedNode.querySelector('iframe.giscus-frame');
          if(giscusIframe) {
            giscusIframe.addEventListener("load", function() {
              window.setTimeout(() => {
                toggleGiscusIfUsed(hasAlternateSentinel(), authorPrefersDark);
              }, 100);
            });
            giscusIframeObserver.disconnect();
          }
        }
      });
    });
  });
  giscusIframeObserver.observe(document.body, { childList: true, subtree: true });
</script>
<input type="hidden" id="giscus-base-theme" value="light">
<input type="hidden" id="giscus-alt-theme" value="dark">
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Copyright 2025, Fabrício Almeida-Silva</p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>