---
title: "syntenet"
description: |
  An R/Bioconductor package for the inference and analysis of
  synteny networks
date: "2022-12-27"
author: 
  - name: Fabr√≠cio Almeida-Silva
    orcid: 0000-0002-5314-2964
draft: false
image: featured_syntenet_logo.png
categories:
  - R packages
  - bioconductor
  - comparative genomics
  - evolutionary genomics
  - networks
  - systems biology
  - bioconductor
---

## Summary

The goal of `syntenet` is to infer synteny networks from whole-genome protein sequence data and analyze them. Anchor pairs from synteny analyses are treated as an undirected unweighted graph (i.e., a synteny network), and users can perform:

-   **Synteny detection** using a native implementation of the [MCScanX algorithm](https://doi.org/10.1093/nar/gkr1293), a C++ program that has been modified and ported to R with Rcpp. This way, users do not need to install MCScanX beforehand, because `syntenet` has its own implementation of the same algorithm.
-   **Synteny network inference** by treating anchor pairs as edges of a graph;
-   **Network clustering** using the Infomap algorithm;
-   **Phylogenomic profiling**, which consists in identifying which species contain which clusters. This analysis can reveal highly conserved synteny clusters and taxon-specific ones (e.g., family- and order-specific clusters);
-   **Microsynteny-based phylogeny reconstruction** with maximum likelihood, which can be achieved by inferring a phylogeny from a binary matrix of phylogenomic profiles with IQTREE.
