<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>My publications â€“ FabrÃ­cio Almeida-Silva</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../img/logo.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-5923429b3b9561efef9f0e8877f8174d.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-9TMPT34HZ4"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-9TMPT34HZ4', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../styles.css">
<meta name="twitter:title" content="FabrÃ­cio Almeida-Silva">
<meta name="twitter:description" content="Hi, Iâ€™m FabrÃ­cio! Welcome to my personal website and blog.">
<meta name="twitter:image" content="https://almeidasilvaf.github.io/publications/img/logo.png">
<meta name="twitter:creator" content="@almeidasilvaf">
<meta name="twitter:site" content="@almeidasilvaf">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed fullcontent quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../img/logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">FabrÃ­cio Almeida-Silva</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../about/index.html"> 
<span class="menu-text">About me</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../software/index.html"> 
<span class="menu-text">Software &amp; Data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../publications/index.html" aria-current="page"> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../teaching/index.html"> 
<span class="menu-text">Teaching</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../blog/index.html"> 
<span class="menu-text">Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://bsky.app/profile/almeidasilvaf.bsky.social"> 
<span class="menu-text"><i class="fa-brands fa-bluesky" aria-label="bluesky"></i></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/almeidasilvaf"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:fabricio_almeidasilva@hotmail.com"> <i class="bi bi-envelope" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">My publications</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="preprints-awaiting-peer-review" class="level2">
<h2 class="anchored" data-anchor-id="preprints-awaiting-peer-review">Preprints awaiting peer review</h2>
<p><strong>Almeida-Silva F</strong>, Van de Peer Y (2025). â€œGene expression divergence following gene and genome duplications in spatially resolved plant transcriptomes.â€ <em>bioRxiv</em>, 2025-05. <a href="https://doi.org/10.1101/2025.05.27.656262" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>Taheri A, <strong>Almeida-Silva F</strong>, Zhang Y, Fu X, Li L, Wang Y, Tang K (2025). â€œArtemisia Database: A Comprehensive Resource for Gene Expression and Functional Insights in Artemisia annua.â€ <em>bioRxiv</em>, 2025-05. <a href="https://doi.org/10.1101/2025.05.21.655314" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>Olvera-Vazquez SG, Chen X, Mesnil A, Meslin C, <strong>Almeida-Silva F</strong>, Confais J, Bourgeois Y, Lougmani C, Alix K, Choisne N, others (2025). â€œComprehensive annotation of olfactory and gustatory receptor genes and transposable elements revealed their evolutionary dynamics in aphids.â€ <em>bioRxiv</em>, 2025-04. <a href="https://doi.org/10.1101/2025.04.14.648604" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>Chen H, <strong>Almeida-Silva F</strong>, Logghe G, Bonte D, Van de Peer Y (2024). â€œThe rise of polyploids during environmental catastrophes.â€ <em>bioRxiv</em>, 2024-11. <a href="https://doi.org/10.1101/2024.11.22.624806" class="citation-link" target="_blank">ğŸ”—</a></p>
</section>
<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section">2025</h2>
<p>26. Zhang R, Shang H, Milne RI, <strong>Almeida-Silva F</strong>, Chen H, Zhou M, Shu H, Jia K, Van de Peer Y, Ma Y (2025). â€œSOI: robust identification of orthologous synteny with the Orthology Index and broad applications in evolutionary genomics.â€ <em>Nucleic Acids Research</em>, <em>53</em>(7), gkaf320. <a href="https://doi.org/10.1093/nar/gkaf320" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>25. Drnevich J, Tan FJ, <strong>Almeida-Silva F</strong>, Castelo R, Culhane AC, Davis S, Doyle MA, Holmes S, Lahti L, Mahmoud A, Nishida K, Ramos M, Rue-Albrecht K, Shih DJH, Gatto L, Soneson C (2025). â€œLearning and teaching biological data science in the Bioconductor community.â€ <em>PLoS Computational Biology</em>. <a href="https://doi.org/10.1371/journal.pcbi.1012925" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>24. <strong>Almeida-Silva F</strong>, Van de Peer Y (2025). â€œdoubletrouble: an R/Bioconductor package for the identification, classification, and analysis of gene and genome duplications.â€ <em>Bioinformatics</em>, btaf043. <a href="https://doi.org/10.1093/bioinformatics/btaf043" class="citation-link" target="_blank">ğŸ”—</a></p>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1">2024</h2>
<p>23. Prost-Boxoen L, Bafort Q, Van de Vloet A, <strong>Almeida-Silva F</strong>, Paing YT, Casteleyn G, Dâ€™hondt S, De Clerck O, Van de Peer Y (2024). â€œAsymmetric genome merging leads to gene expression novelty through nucleo-cytoplasmic disruptions and transcriptomic shock in Chlamydomonas triploids.â€ <em>New Phytologist</em>. <a href="https://doi.org/10.1111/nph.20249" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>22. Barbosa-Xavier K, Pedrosa-Silva F, <strong>Almeida-Silva F</strong>, Venancio TM (2024). â€œCannabis Expression Atlas: a comprehensive resource for integrative analysis of Cannabis sativa L. gene expression.â€ <em>Physiologia Plantarum</em>, <em>176</em>(6), e70010. <a href="https://doi.org/10.1111/ppl.70010" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>21. Turquetti-Moraes DK, Cardoso-Silva CB, <strong>Almeida-Silva F</strong>, Venancio TM (2024). â€œMultiomic analysis of genes related to oil traits in legumes provide insights into lipid metabolism and oil richness in soybean.â€ <em>Plant Physiology and Biochemistry</em>, 109180. ISSN 0981-9428, <a href="https://doi.org/10.1016/j.plaphy.2024.109180" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>20. Wu T, Bafort Q, Mortier F, <strong>Almeida-Silva F</strong>, Natran A, Van de Peer Y (2024). â€œThe immediate metabolomic effects of whole-genome duplication in the greater duckweed, Spirodela polyrhiza.â€ <em>American Journal of Botany</em>, e16383. <a href="https://doi.org/10.1002/ajb2.16383" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>19. <strong>Almeida-Silva F</strong>, Prost-Boxoen L, Van de Peer Y (2024). â€œhybridexpress: an R/Bioconductor package for comparative transcriptomic analyses of hybrids and their progenitors.â€ <em>New Phytologist</em>, <em>243</em>(2), 811-819. <a href="https://doi.org/10.1111/nph.19862" class="citation-link" target="_blank">ğŸ”—</a></p>
</section>
<section id="section-2" class="level2">
<h2 class="anchored" data-anchor-id="section-2">2023</h2>
<p>18. Pinto VB, Vidigal PMP, Dal-Bianco M, <strong>Almeida-Silva F</strong>, Venancio TM, Viana JMS (2023). â€œTranscriptome-based strategies for identifying aluminum tolerance genes in popcorn (Zea mays L. var. everta).â€ <em>Scientific Reports</em>, <em>13</em>(1), 19400. <a href="https://doi.org/10.1038/s41598-023-46810-9" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>17. Cherene MB, Taveira GB, <strong>Almeida-Silva F</strong>, da Silva MS, Cavaco MC, da Silva-Ferreira AT, Perales JEA, de Oliveira Carvalho A, VenÃ¢ncio TM, da Motta OV, others (2023). â€œStructural and Biochemical Characterization of Three Antimicrobial Peptides from Capsicum annuum L. var. annuum Leaves for Anti-Candida Use.â€ <em>Probiotics and Antimicrobial Proteins</em>, 1-18. <a href="https://doi.org/10.1007/s12602-023-10112-3" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>16. Cao Y, <strong>Almeida-Silva F</strong>, Zhang W, Ding Y, Bai D, Bai W, Zhang B, Van de Peer Y, Zhang D (2023). â€œGenomic Insights into Adaptation to Karst Limestone and Incipient Speciation in East Asian Platycarya spp.(Juglandaceae).â€ <em>Molecular Biology and Evolution</em>, <em>40</em>(6), msad121. <a href="https://doi.org/10.1093/molbev/msad121" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>15. <strong>Almeida-Silva F</strong>, Venancio TM (2023). â€œDiscovering and prioritizing candidate resistance genes against soybean pests by integrating GWAS and gene coexpression networks.â€ <em>Gene</em>, <em>860</em>, 147231. <a href="https://doi.org/10.1016/j.gene.2023.147231" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>14. <strong>Almeida-Silva F</strong>, Pedrosa-Silva F, Venancio TM (2023). â€œThe Soybean Expression Atlas v2: A comprehensive database of over 5000 RNA-seq samples.â€ <em>The Plant Journal</em>. <a href="https://doi.org/10.1111/tpj.16459" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>13. <strong>Almeida-Silva F</strong>, Van de Peer Y (2023). â€œAssessing the quality of comparative genomics data and results with the cogeqc R/Bioconductor package.â€ <em>Methods in Ecology and Evolution</em>. <a href="https://doi.org/10.1111/2041-210X.14243" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>12. <strong>Almeida-Silva F</strong>, Van de Peer Y (2023). â€œWhole-genome Duplications and the Long-term Evolution of Gene Regulatory Networks in Angiosperms.â€ <em>Molecular Biology and Evolution</em>, <em>40</em>(7), msad141. <a href="https://doi.org/10.1093/molbev/msad141" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>11. <strong>Almeida-Silva F</strong>, Zhao T, Ullrich KK, Schranz ME, Van de Peer Y (2023). â€œsyntenet: an R/Bioconductor package for the inference and analysis of synteny networks.â€ <em>Bioinformatics</em>, <em>39</em>(1), btac806. <a href="https://doi.org/10.1093/bioinformatics/btac806" class="citation-link" target="_blank">ğŸ”—</a></p>
</section>
<section id="section-3" class="level2">
<h2 class="anchored" data-anchor-id="section-3">2022</h2>
<p>10. Irineu LESdS, Soares CdP, Soares TS, Almeida FAd, <strong>Almeida-Silva F</strong>, Gazara RK, Meneses CHSG, Canellas LP, Silveira V, Venancio TM, others (2022). â€œMultiomic approaches reveal hormonal modulation and nitrogen uptake and assimilation in the initial growth of maize inoculated with Herbaspirillum seropedicae.â€ <em>Plants</em>, <em>12</em>(1), 48. <a href="https://doi.org/10.3390/plants12010048" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>9. <strong>Almeida-Silva F</strong>, Venancio TM (2022). â€œBioNERO: an all-in-one R/Bioconductor package for comprehensive and easy biological network reconstruction.â€ <em>Functional &amp; Integrative Genomics</em>, <em>22</em>(1), 131-136. <a href="https://doi.org/10.1007/s10142-021-00821-9" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>8. <strong>Almeida-Silva F</strong>, Venancio TM (2022). â€œPathogenesis-related protein 1 (PR-1) genes in soybean: Genome-wide identification, structural analysis and expression profiling under multiple biotic and abiotic stresses.â€ <em>Gene</em>, <em>809</em>, 146013. <a href="https://doi.org/10.1016/j.gene.2021.146013" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>7. Turquetti-Moraes DK, Moharana KC, <strong>Almeida-Silva F</strong>, Pedrosa-Silva F, Venancio TM (2022). â€œIntegrating omics approaches to discover and prioritize candidate genes involved in oil biosynthesis in soybean.â€ <em>Gene</em>, <em>808</em>, 145976. <a href="https://doi.org/10.1016/j.gene.2021.145976" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>6. <strong>Almeida-Silva F</strong>, Venancio TM (2022). â€œcageminer: an R/Bioconductor package to prioritize candidate genes by integrating genome-wide association studies and gene coexpression networks.â€ <em>in silico Plants</em>, <em>4</em>(2), diac018. <a href="https://doi.org/10.1093/insilicoplants/diac018" class="citation-link" target="_blank">ğŸ”—</a></p>
</section>
<section id="section-4" class="level2">
<h2 class="anchored" data-anchor-id="section-4">2021</h2>
<p>5. <strong>Almeida-Silva F</strong>, Venancio TM (2021). â€œIntegration of genome-wide association studies and gene coexpression networks unveils promising soybean resistance genes against five common fungal pathogens.â€ <em>Scientific Reports</em>, <em>11</em>(1), 24453. <a href="https://doi.org/10.1038/s41598-021-03864-x" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>4. Sangi S, AraÃºjo PM, Coelho FS, Gazara RK, <strong>Almeida-Silva F</strong>, Venancio TM, Grativol C (2021). â€œGenome-wide analysis of the COBRA-Like gene family supports gene expansion through Whole-Genome Duplication in soybean (Glycine max).â€ <em>Plants</em>, <em>10</em>(1), 167. <a href="https://doi.org/10.3390/plants10010167" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>3. <strong>Almeida-Silva F</strong>, Moharana KC, Venancio TM (2021). â€œThe state of the art in soybean transcriptomics resources and gene coexpression networks.â€ <em>in silico Plants</em>, <em>3</em>(1), diab005. <a href="https://doi.org/10.1093/insilicoplants/diab005" class="citation-link" target="_blank">ğŸ”—</a></p>
</section>
<section id="section-5" class="level2">
<h2 class="anchored" data-anchor-id="section-5">2020</h2>
<p>2. <strong>Almeida-Silva F</strong>, Moharana KC, Machado FB, Venancio TM (2020). â€œExploring the complexity of soybean (Glycine max) transcriptional regulation using global gene co-expression networks.â€ <em>Planta</em>, <em>252</em>, 1-12. <a href="https://doi.org/10.1007/s00425-020-03499-8" class="citation-link" target="_blank">ğŸ”—</a></p>
<p>1. Machado FB, Moharana KC, <strong>Almeida-Silva F</strong>, Gazara RK, Pedrosa-Silva F, Coelho FS, Grativol C, Venancio TM (2020). â€œSystematic analysis of 1298 RNA-Seq samples and construction of a comprehensive soybean (Glycine max) expression atlas.â€ <em>The Plant Journal</em>, <em>103</em>(5), 1894-1909. <a href="https://doi.org/10.1111/tpj.14850" class="citation-link" target="_blank">ğŸ”—</a></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/almeidasilvaf\.github\.io\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Copyright 2025, FabrÃ­cio Almeida-Silva</p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>